compiler/rustc_typeck/src/check/generator_interior.rs

```diff
@@ -210,6 +224,40 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
         NestedVisitorMap::None
     }
 
+    fn visit_arm(&mut self, arm: &'tcx Arm<'tcx>) {
+        if arm.guard.is_some() {
+            self.guard_bindings.push(<_>::default());
+        }
```

Comment:
This is duplicated below, I think maybe you meant to take it out?

Reply:
Yes. I need to remove it.